detections/network/http_c2_framework_user_agent.yml

```diff
@@ -0,0 +1,81 @@
+name: HTTP C2 Framework User Agent
+id: 229dc225-6abe-4d28-89fd-edf874086162
+version: 1
+date: '2025-12-15'
+author: Ravent Tait, Splunk
+status: production
+type: TTP
+description: This Splunk query analyzes web logs to identify and categorize 
+  user agents, detecting various types of c2 frameworks. This activity can signify malicious actors
+  attempting to interact with hosts on the network using known default configurations of command
+  and control tools.
+data_source:
+- Splunk Stream HTTP
+- Suricata
+search: '| tstats  `security_content_summariesonly` count min(_time) as firstTime max(_time) as lastTime
+  from datamodel=Web by Web.http_user_agent Web.http_method, Web.url, Web.url_length Web.src, Web.dest 
+  | `drop_dm_object_name("Web")`
+  | lookup suspicious_c2_user_agents c2_user_agent AS http_user_agent OUTPUT tool, description
+  | where isnotnull(tool)
+  | stats count min(firstTime) as first_seen max(lastTime) as last_seen
+  by tool url http_user_agent src dest description
+  | `security_content_ctime(first_seen)` | `security_content_ctime(last_seen)`
+  | `http_c2_framework_user_agent_filter`'
+how_to_implement: To successfully implement this search, you need to be ingesting
+  web or proxy logs, or ensure it is being filled by a proxy like device, into the
+  Web Datamodel. For additional filtering, allow list private IP space or restrict
+  by known good.
+known_false_positives: Filtering may be required in some instances depending on legacy system usage, filter as needed.
+references:
+- https://github.com/BC-SECURITY/Malleable-C2-Profiles
+- https://www.keysight.com/blogs/en/tech/nwvs/2021/07/28/koadic-c3-command-control-decoded
+- https://github.com/mthcht/awesome-lists/blob/main/Lists/suspicious_http_user_agents_list.csv
+drilldown_searches:
+- name: View the detection results for - "$src$"
+  search: '%original_detection_search% | search  src = "$src$"'
+  earliest_offset: $info_min_time$
+  latest_offset: $info_max_time$
+- name: View risk events for the last 7 days for - "$src$"
+  search: '| from datamodel Risk.All_Risk | search normalized_risk_object IN ("$src$")
+    starthoursago=168  | stats count min(_time) as firstTime max(_time) as lastTime
+    values(search_name) as "Search Name" values(risk_message) as "Risk Message" values(analyticstories)
+    as "Analytic Stories" values(annotations._all) as "Annotations" values(annotations.mitre_attack.mitre_tactic)
+    as "ATT&CK Tactics" by normalized_risk_object | `security_content_ctime(firstTime)`
+    | `security_content_ctime(lastTime)`'
+  earliest_offset: $info_min_time$
+  latest_offset: $info_max_time$
+rba:
+  message: A known C2 Framework user agent $http_user_agent$ was performing a request from $src$ to $dest$.
+  risk_objects:
+  - field: dest
```

Comment:
i think in this case, the src machine is likely the Victim machine connecting to dest, hence the risk_object would only be src and we can remove dest from risk_object and consider adding dest to threat_object of type ip_address

and src risk_object would also be an ip address in Web logs